### PR TITLE
Change `env` defaulting for `Module` to be more robust.

### PR DESCRIPTION
**Pull request by rohinb2**

### CODE REVIEW

**Comment by rohinb2** [comment]
* **#1038** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1038?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/module.py [hunk 1]
```diff
@@ -49,6 +49,24 @@
 ]
 
 
+def _env_for_module(passed_in_env=None, system=None):
```
Comment (dongreenberg) [inline runhouse/resources/module.py:52]: Can this be named more clearly?

File: runhouse/resources/module.py [hunk 2]
```diff
@@ -49,6 +49,24 @@
...
+    active_env_servlet_env = obj_store.get_process_env()
+    if active_env_servlet_env:
+        return active_env_servlet_env
+
+    if system and isinstance(system, Cluster):
+        return system.default_env
+
+    potential_default_env = _get_env_from(_default_env_if_on_cluster())
+    if potential_default_env:
+        return potential_default_env
```
Comment (dongreenberg) [inline runhouse/resources/module.py:65]: I don't think this is accurate - if no system is set and the obj_store doesn't have storage, we shouldn't even put it in the default env, just treat it the same as if created outside of the cluster entirely.

File: runhouse/resources/module.py [hunk 3]
```diff
@@ -49,6 +49,24 @@
...
+
+    return Env()
```
Comment (dongreenberg) [inline runhouse/resources/module.py:67]: I'm not sure about this either - if we don't have an env to infer, I think we should just fall back to the cluster constructor's behavior, not set a default or placeholder.

**Pull request closed without merging**